lib/http/HttpServer.ts

```diff
@@ -54,35 +54,43 @@ class HttpServer {
       return;
     }
 
-    let resJson: any;
-    try {
-      switch (req.url) {
-        case '/resolveraiden':
-          resJson = await this.httpService.resolveHashRaiden(reqJson);
-          break;
-        default:
-          res.writeHead(404);
-          res.end();
-          break;
-      }
-      res.writeHead(200, { 'Content-Type': 'application/json' });
-      res.end(JSON.stringify(resJson));
-    } catch (err) {
-      switch (err.code) {
-        case swapErrorCodes.INVALID_RESOLVE_REQUEST:
-        case errorCodes.INVALID_ARGUMENT:
-          res.writeHead(400);
-          res.end(err.message);
-          break;
-        case swapErrorCodes.PAYMENT_HASH_NOT_FOUND:
-          res.writeHead(404);
-          res.end(err.message);
-          break;
-        default:
-          res.writeHead(500);
-          res.end(JSON.stringify(err));
-          break;
-      }
+    switch (req.url) {
+      case '/resolveraiden':
+        try {
+          const resJson = await this.httpService.resolveHashRaiden(reqJson);
+          res.writeHead(200, { 'Content-Type': 'application/json' });
+          res.end(JSON.stringify(resJson));
+        } catch (err) {
+          switch (err.code) {
+            case swapErrorCodes.INVALID_RESOLVE_REQUEST:
+            case errorCodes.INVALID_ARGUMENT:
+              res.writeHead(400);
+              res.end({ message: err.message, retry: false });
```

Comment:
What is the called expected to do?

Reply:
I refactored this a bit but the functionality is the same. When the resolve request is invalid (e.g. it does not specify a `secrethash` or is otherwise malformed) we respond with `400 BAD REQUEST`.